x-pack/plugins/security_solution/public/assistant/get_comments/custom_codeblock/esql_code_block.tsx

```diff
@@ -0,0 +1,336 @@
+    return getESQLAdHocDataview(indexPattern, dataViewService).then((dataView) => {
+      if (dataView.fields.getByName('@timestamp')?.type === 'date') {
+        dataView.timeFieldName = '@timestamp';
+      }
+      return dataView;
+    });
+  }, [indexPattern, dataViewService]);
+
+  const [isSaveModalOpen, setIsSaveModalOpen] = useState(false);
+  const [isTableVisible, setIsTableVisible] = useState(false);
+  const [lensInput, setLensInput] = useState<TypedLensByValueInput | undefined>(undefined);
+
+  const preferredChartType = undefined; // 'XY';
+
+  // initialization
+  useEffect(() => {
+    if (lensHelpersAsync && dataViewAsync && !lensInput && formattedColumns) {
+      const context = {
+        dataViewSpec: dataViewAsync?.toSpec(),
+        fieldName: '',
+        textBasedColumns: formattedColumns,
+        query: {
+          esql: value,
+        },
+      };
+
+      const chartSuggestions = lensHelpersAsync.suggestions(
+        context,
+        dataViewAsync,
+        [],
+        preferredChartType
+      );
+
+      if (chartSuggestions?.length) {
+        const [suggestion] = chartSuggestions;
+
+        const attrs = getLensAttributesFromSuggestion({
+          filters: [],
+          query: {
+            esql: value,
+          },
+          suggestion,
+          dataView: dataViewAsync,
+        }) as TypedLensByValueInput['attributes'];
+
+        const lensEmbeddableInput = {
+          attributes: attrs,
+          id: generateId(),
+        };
+        setLensInput(lensEmbeddableInput);
+      }
+    }
+  }, [dataViewAsync, formattedColumns, lensHelpersAsync, lensInput, preferredChartType, value]);
+
+  // if the Lens suggestions api suggests a table then we want to render a Discover table instead
+  const isLensInputTable = lensInput?.attributes?.visualizationType === 'lnsDatatable';
+
+  const handleShowVisualization = useCallback(() => setShowVisualization(true), []);
+
+  const handleToogleTableView = useCallback(() => setIsTableVisible((prev) => !prev), []);
+
+  const handleCloseSaveModal = useCallback(() => setIsSaveModalOpen(false), []);
+
+  return (
+    <>
+      <EuiPanel
+        hasShadow={false}
+        hasBorder={false}
+        paddingSize="s"
+        className={css`
+          background-color: ${theme.euiTheme.colors.lightestShade};
+          .euiCodeBlock__pre {
+            margin-bottom: 0;
+            padding: ${theme.euiTheme.size.m};
+            min-block-size: 48px;
+          }
+          .euiCodeBlock__controls {
+            inset-block-start: ${theme.euiTheme.size.m};
+            inset-inline-end: ${theme.euiTheme.size.m};
+          }
+        `}
+      >
+        <EuiFlexGroup direction="column" gutterSize="xs">
+          <EuiFlexItem grow={false}>
+            <EuiCodeBlock isCopyable fontSize="m" language="sql">
+              {value}
+            </EuiCodeBlock>
+          </EuiFlexItem>
+
+          {!showVisualization && (
```

Comment:
If it relies on `knowledge base` to work, should we check if the knowledge base toggle is on here?
<img width="787" alt="Screenshot 2024-07-04 at 17 23 52" src="https://github.com/elastic/kibana/assets/6295984/dc7b0349-777b-4938-95ba-cf59d29dacef">

Reply:
Thank you @angorayc, the UI part doesn't need Knowledge base to be enabled, but it requires in your case to `packetbeat-*` to exists and have proper data propagated

Reply:
Yes, I can confirm that I have packetbeat-* data. I searched the same ESQL within timeline, and there was one entry appeared in the table, but it wasn't able to be visualised from the case markdown when using `preview`.

Reply:
will double check, Thank you 🙇